crates/shared/src/maintenance.rs

```diff
@@ -2,20 +2,48 @@ use crate::current_block::{self, BlockInfo, CurrentBlockStream};
 use anyhow::{ensure, Result};
 use futures::{future::join_all, Stream, StreamExt as _};
 use std::{sync::Arc, time::Duration};
+use strum::IntoEnumIterator;
 use tokio::time;
 use tracing::Instrument as _;
 
+/// List of all possible maintainers in the system
+#[derive(strum::EnumIter)]
+pub enum Maintainer {
+    ServiceMaintenance,
+    EventHandler,
+    EventUpdater,
+    Postgres,
+    BalancerPoolFetcher,
+    UniswapV2PoolFetcher,
+    UniswapV3PoolFetcher,
+}
+
+impl Maintainer {
+    fn label(&self) -> &'static str {
+        match self {
+            Maintainer::ServiceMaintenance => "ServiceMaintenance",
```

Comment:
`strum` also offers a macro to generate this boiler plate code.

Reply:
Didn't know. Fixed.